python/cudf/udf_cpp/strings/src/strings/udf/udf_apis.cu

```diff
@@ -98,6 +129,24 @@ void free_udf_string_array(cudf::strings::udf::udf_string* d_strings,
                      [d_strings] __device__(auto idx) { d_strings[idx].clear(); });
 }
 
+/**
+ * @copydoc free_managed_udf_string_array
+ *
+ * @param stream CUDA stream used for allocating/copying device memory and launching kernels
+ */
+void free_managed_udf_string_array(cudf::strings::udf::managed_udf_string* managed_strings,
+                                   cudf::size_type size,
+                                   rmm::cuda_stream_view stream)
+{
+  thrust::for_each_n(rmm::exec_policy(stream),
+                     thrust::make_counting_iterator(0),
+                     size,
+                     [managed_strings] __device__(auto idx) {
+                       managed_strings[idx].udf_str.clear();
+                       free(managed_strings[idx].meminfo);
```

Comment:
Would this have been allocated with `NRT_MemInfo_new`? If so, should it be freed with a call to `NRT_MemInfo_destroy` - this would keep symmetry in the API usage, and ensure the alloc / free stats balance.

Reply:
this is a great suggestion and I'll make this change.